buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/converter.py

```diff
@@ -0,0 +1,179 @@
+                    ids=unit_ids,
+                    key="cell_type",
+                    values=[str(celltype_mapping[str(x[0])]) for x in celltype_info["label"][0][0][0]],
+                )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        session_path = Path(self.source_data["file_path"]).parent
+        session_id = session_path.stem
+        # TODO: add condition for retrieving ecephys metadata if no recording or lfp are included in conversion
+        metadata["NWBFile"].update(session_id=session_id)
+
+        unit_properties = []
+        cellinfo_file_path = session_path / f"{session_id}.spikes.cellinfo.mat"
+        if cellinfo_file_path.is_file():
+            cell_info_fields = self.cell_info_fields
+            if "cluID" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="clu_id",
+                        description="0-indexed id of cluster identified from the shank.",
+                    )
+                )
+            if "shankID" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="group_id",
+                        description="The electrode group ID that each unit was identified by.",
+                    )
+                )
+            if "region" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="location",
+                        description="Brain region where each unit was detected.",
+                    )
+                )
+        celltype_filepath = session_path / f"{session_id}.CellClass.cellinfo.mat"
+        if celltype_filepath.is_file():
+            celltype_info = scipy.io.loadmat(celltype_filepath).get("CellClass", np.empty(0))
+            if "label" in celltype_info.dtype.names:
+                unit_properties.append(
+                    dict(
+                        name="cell_type",
+                        description="Type of cell this has been classified as.",
+                    )
+                )
+        metadata.update(Ecephys=dict(UnitProperties=unit_properties))
+        return metadata
+
+
+class HuzsarNWBConverter(NWBConverter):
+    """Primary conversion class for the Huzsar hippocampus data set."""
+
+    data_interface_classes = dict(
+        Behavior=HuzsarBehaviorInterface,
+        Sorting=CellExplorerSortingInterface,
+    )
+
+    def __init__(self, source_data: dict, verbose: bool = True):
+        super().__init__(source_data=source_data, verbose=verbose)
+
+    def get_metadata(self):
+        # metadata = super().get_metadata()
+        # metadata["NWBFile"].update(session_start_time=session_start, session_id=session_id)
+        # metadata.update(Subject=dict(subject_id=subject_id))
+
+        metadata = super().get_metadata()
+        import datetime as dt
+
+        # Get today date
+        date = dt.date.today()
```

Comment:
Timezone?

Reply:
Done now. The specific time is not available though.

Reply:
Hmm.. The `session.mat` does not have the session time somewhere in it? and these folder path names are not specifying any format of date?

Reply:
Not clear, check it out:

https://gist.github.com/h-mayorquin/ea6b6bc1b1b198ebbcb9eba434ce8bbc

You can find the folder name in `basePath` and the folder path names does not seem to match with the date in the same file. Maybe the epochs there could be the key but I need to dig deeper.

Reply:
I interpret the last 6 digits of the base path as follows: March 3rd, 2021

So a date for the recording session but not exact time, fine to do the usual and start from midnight `0`

This is further reinforced by the folder of the kilosort ('spikeSorting' -> 'relativePath') which has an exact datetime of March 5th, 2021 at 11pm:06:45 since the spike sorting would have been run offline after the fact (several days after not uncommon)